pkg/stat/metric/rolling_policy.go

```diff
@@ -65,7 +65,7 @@ func (r *RollingPolicy) add(f func(offset int, val float64), val float64) {
 			offset = i
 		}
 		r.offset = offset
-		r.lastAppendTime = time.Now()
+		r.lastAppendTime = r.lastAppendTime.Add(time.Duration(timespan * int(r.bucketDuration)))
```

Comment:
if time.since(r.lastAppendTime) > r.size * r.bucketDuration, r.timespan returns r.size, not real time span.
maybe we should replace timespan with `time.since(r.lastAppendTime) / r.bucketDuration`.

Reply:
another solution, returns the real time span in func `timespan`, and changes the logic of calculating the variable `e`.

Reply:
I prefer that func `timespan` returns the real time span, because it is more accurate to use `time.Since` once when `RollingPolicy.Add` called. Some test cases were supplied. Thanks for your review~